src/GrasshopperTeklaDrawingLink/Components/Text/CreateTextComponent.cs

```diff
@@ -0,0 +1,263 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.Linq;
+
+using Grasshopper.Kernel;
+
+using GTDrawingLink.Extensions;
+using GTDrawingLink.Tools;
+using GTDrawingLink.Types;
+
+using Rhino.Geometry;
+
+using Tekla.Structures.Drawing;
+
+using TSD = Tekla.Structures.Drawing;
+
+namespace GTDrawingLink.Components.Text {
+    public class CreateTextComponent : CreateDatabaseObjectComponentBase {
+        public override GH_Exposure Exposure => GH_Exposure.primary;
+        protected override Bitmap Icon => Properties.Resources.Text;
+
+        public CreateTextComponent() : base(ComponentInfos.CreateTextComponent) {
+        }
+        protected override void RegisterInputParams(GH_InputParamManager pManager) {
+            pManager.AddParameter(new TeklaDatabaseObjectParam(ParamInfos.View, GH_ParamAccess.item));
+         
+            AddPointParameter(pManager, ParamInfos.MarkInsertionPoint, GH_ParamAccess.list);
+            AddPointParameter(pManager, ParamInfos.MarkLeaderLineEndPoint, GH_ParamAccess.list, true);
+            AddTextParameter(pManager, ParamInfos.Text, GH_ParamAccess.list, true);
+            SetParametersAsOptional(pManager, new List<int> {
```

Comment:
Two parameters above and one below are also optional. Why are they not wrapped into this helper method (_SetParametersAsOptional_)?